pallets/settlement/src/lib.rs

```diff
@@ -1340,50 +1394,105 @@ impl<T: Trait> Module<T> {
+                                if let Ok(from_mercat_pending_state) = Self::get_pending_state(
+                                    &leg.from.did,
+                                    &leg.from_account_id,
+                                    instruction_id,
+                                ) {
                                     if let Ok((to_mercat, to_mercat_balance)) =
                                         Self::get_account_and_balance(
                                             &leg.to.did,
                                             &leg.to_account_id,
                                         )
                                     {
-                                        // Verify the proofs.
-                                        let mut rng = rng::Rng::default();
-                                        let result = TransactionValidator {}.verify_transaction(
-                                            &decoded_justified_tx,
-                                            &from_mercat,
-                                            &from_mercat_balance,
-                                            &to_mercat,
-                                            &to_mercat_balance,
-                                            &[],
-                                            &mut rng,
-                                        );
-                                        if let Ok((new_sender_balance, new_receiver_balance)) =
-                                            result
-                                        {
-                                            <ConfidentialAsset<T>>::update_mercat_account_balance(
-                                                &leg.from.did,
-                                                &leg.from_account_id,
-                                                new_sender_balance,
-                                            );
-                                            <ConfidentialAsset<T>>::update_mercat_account_balance(
-                                                &leg.to.did,
-                                                &leg.to_account_id,
-                                                new_receiver_balance,
-                                            );
+                                        // Get sender's current balance.
+                                        if let Ok((from_mercat, _)) = Self::get_account_and_balance(
+                                            &leg.from.did,
+                                            &leg.from_account_id,
+                                        ) {
+                                            // Verify the proofs.
+                                            let mut rng = rng::Rng::default();
+                                            let result = TransactionValidator {}
+                                                .verify_transaction(
+                                                    &decoded_justified_tx,
```

Comment:
This is *several* levels too indented and so the function needs to be split up to make things readable.

Reply:
Turns out there's a already a task to address this: https://polymath.atlassian.net/browse/MESH-1329.
I tried to get rid of the unnecessary indentations as much as possible.